### PR TITLE
[REVIEW] Add file tags to libcudf doxygen

### LINKED ISSUE #5965: [FEA] Header file information on APIs in libcudf doc page
**Is your feature request related to a problem? Please describe.**
Currently we don't have information about header file from which the API has been originated, it would be great to have this information. 
https://docs.rapids.ai/api/libcudf/nightly/group__column__copy.html

### PR DESCRIPTION
**Pull request by davidwendt**
Closes #5965 
The current doxygen rendering of libcudf does not include information on what headers file the functions are declared. Once you have found that cool API you want to use, you need to hunt for it in the `cudf/include` directory. Some files are rendered by doxygen automatically -- usually when declared as classes in a `.hpp` file. But C-style function APIs (which is most of libcudf) and classes defined in `.cuh` do not get rendered.

Doxygen supports a `@file` command that can be used to include the name of the file where the declarations are found. This picture shows the header file highlighted as a link.
doxygen-with-file2

Clicking on the link shows a doxygen rendering of everything in that file:
doxygen-with-file-sum

Furthermore, doxygen also has a nice _raw_ rendering of the header file without any doxygen comments when clicking on the "Go to the source code of this file" link.
doxygen-with-file-raw

This PR contains alot of files but most are just adding the `@file` tag to a doxygen comment. There is no code change and only small modification to the `Doxyfile` configuration file that allows `.cuh` files to be rendered like `.hpp` ones.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/include/cudf/strings/split/partition.hpp [hunk 1]
```diff
@@ -24,6 +24,8 @@ namespace strings {
 /**
  * @addtogroup strings_split
  * @{
+ * @file strings/split/partition.hpp
```
Comment (vuule) [inline cpp/include/cudf/strings/split/partition.hpp:27]: why is the file name here, and not in other files?

**Review by vuule — COMMENTED** [review]
Looks good, minor comments:

File: cpp/include/cudf/copying.hpp [hunk 2]
```diff
@@ -29,6 +29,8 @@ namespace cudf {
 /**
  * @addtogroup column_copy
  * @{
+ * @file
+ * @brief Column APIs for gather, scatter, split, slice, etc.
```
[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/include/cudf/copying.hpp:33]: You are right. Here is a snippet of where it should've appeared in the screen capture.
doxygen-file-brief
Also, they will appear in the [Files](https://docs.rapids.ai/api/libcudf/stable/files.html) list

File: cpp/include/cudf/strings/split/partition.hpp (hunk 1 above)
Comment (davidwendt) [reply cpp/include/cudf/strings/split/partition.hpp:27]: The name is optional but if there are two files with the same name in different directories (we have a few) then specifying the name can help distinguish them. I found that doxygen is ignoring my suggestions but I left some of these in anyway.

**Review by vuule — APPROVED** [review]

**Comment by vuule** [comment]
rerun tests

**Review by harrism — APPROVED** [review]

**Pull request merged**